File: Unidad_4/paramether_pass/param_pass.py

def funcion(a):
    a = a + 1
    return a

def funcionList(l):
    l.append(3)
    return l

a,b = 0,0
b = funcion(a)

print a, b

#print id(a), id(b)

l = []
m = []
l.append(1)
l.append(2)
m.append(0)

m = funcionList(l)

#print id(l), id(m)
m.append(4)
print l, m

File: Unidad_4/paramether_pass/param_pass.c
#include <stdio.h>

int funcion(int a, int *b){
  a+=1;
  *b += 1;
  int* c = &a;
  return *c;
}

int main(){
  int a = 0, b = 0, c;
  c = funcion(a,&b);
  printf("%d %d %d",a,b,c);
  return 0;
}
